### PR TITLE
Add Activity item to diagram

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Activity elements can't be added to a diagram

Issue Number: #974 

### What is the new behavior?

Activities can be added to a diagram.

- [x] ActivityParameterNode can be added via Activity property page
- [x] ActivityParameterNode can not be grouped/ungrouped in diagrams
- [x] ActivityParameterNode and Parameter can't be moved in tree view
- [x] ActivityParameterNode is always connected to the edge of an Activity
- [x] ActivityParameterNode can't be copied/pasted individually
- [x] Copy/paste copies Activity with ActivityParameterNode's
- [x] In-diagram editing of parameter names
- [x] Icons for ActivityParameterNode and Parameter
- [x] Connect to Parameter nodes with Object flow

### Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### Other information

- [x] Any `AttachedPresentation` can more easily be dragged: handle is disconnect on drag.

* Parameters are created specifically for the ActivityParameterNode. They're not shared.
* ActivityParameterNode's in tree view should show Parameter name?
* Should Gaphor allow for nested Activities?

### CODE REVIEW

**Comment by danyeaw** [comment]
I am getting a fish effect on the sides of the activity shape:
image

How do I add an Activity Parameter Node to an activity? It says in the description that it is through the property page, but when I click on the activity only the Name, Is Abstract, and Notes shows up:
image

**Comment by amolenaar** [comment]
Odd. You should see the parameters like this:
image

**Comment by danyeaw** [comment]
I did pull down the latest push you made and ran `poetry install` again, things are working now. :+1: 

This looks really nice for the resnapping back to the border, and the parameter nodes move around nicely. When the Parameter Nodes are selected, the red square is right on top of the name inside the parameter node. Do we want to allow users to change the size of them? What if the text is too large to fit inside the parameter node?

I added some more details on how I would expect activities to work longer term in #974.

**Comment by amolenaar** [comment]
> [...] When the Parameter Nodes are selected, the red square is right on top of the name inside the parameter node. 

With the current logic I need a handle to connect. Handles do not have a "visible" flag, although it would be pretty easy to add. Would it be better if the handle was hidden?

> Do we want to allow users to change the size of them?

I did not add that, to keep those elements more in line with proxy port and (later) pins.

> What if the text is too large to fit inside the parameter node?

The parameter node will currently resize, but text will not wrap (same as Packages and Classes). Would this pose a problem?

**Comment by danyeaw** [comment]
@amolenaar Would it be difficult to show the handles at the top and bottom of the Parameter Node?

**Comment by amolenaar** [comment]
@danyeaw ehhmm. Yes. The handle is used to connect the element.
If Handles are located at the top/bottom, parameters attached to the top of the activity would look odd.

Why do you want to have them at the top/bottom?

**Comment by danyeaw** [comment]
@amolenaar I was just thinking of some type of solution to avoid putting the handle in the middle of the label.

**Comment by amolenaar** [comment]
Okay. We can make the handle hidden. Although you can not see if it's selected then 🤔

**Review by danyeaw — APPROVED** [review]
The implementation looks really good!

File: gaphor/SysML/blocks/proxyport.py [hunk 1]
```diff
@@ -32,45 +25,10 @@ def text_position(position):
 
 
 @represents(sysml.ProxyPort)
-class ProxyPortItem(Named, Presentation[sysml.ProxyPort], HandlePositionUpdate):
+class ProxyPortItem(Named, AttachedPresentation[sysml.ProxyPort]):
```
[the remark this answers is not in the document]
Comment (amolenaar) [reply gaphor/SysML/blocks/proxyport.py:28]: Thanks :smiley:

**Pull request merged**